### PR TITLE
Diff viewer convert build api markdown documenter.yml pipeline to 1 es template

### PR DESCRIPTION
**Pull request by seanimam**
### DO NOT MERGE

 This PR was created to clearly demonstrate what the diff between the new copies of pipeline templates within the 1ES folder actually are within this PR https://github.com/microsoft/FluidFramework/pull/19243

### CODE REVIEW

File: tools/pipelines/templates/1ES/include-policy-check.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        - task: Npm@1
+          displayName: npm run ${{ check }}
+          inputs:
+            command: 'custom'
+            workingDir: ${{ parameters.buildDirectory }}
+            customCommand: 'run ${{ check }}'
+          condition: succeededOrFailed()
+
+      - task: Bash@3
+        displayName: Check for extraneous modified files
```
Comment (tylerbutler) [inline tools/pipelines/templates/1ES/include-policy-check.yml:60]: Probably want to externalize this script or make it a template itself. Not blocking this PR.

File: tools/pipelines/templates/1ES/include-policy-check.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+# Licensed under the MIT License.
+
+# include-policy-check
+#
+# This template can be included in pipelines that run check scripts as part of their pipeline. The check scripts are
+# executed serially in a separate stage from the main build.
+#
+# IMPORTANT: This template only works with pnpm. Don't use in any pipelines that use another package manager.
```
[the remark this answers is not in the document]
Comment (alexvy86) [reply tools/pipelines/templates/1ES/include-policy-check.yml:9]: It's just copied from the current version of the file so I'd say not touch it in this PR. I think the original comment is yours @tylerbutler 😄 .

File: tools/pipelines/templates/build-npm-package.yml [hunk 3]
```diff
@@ -4,601 +4,645 @@
...
+                      targetPath: $(Build.ArtifactStagingDirectory)/_api-extractor-temp
+                      artifactName: _api-extractor-temp
+                      sbomEnabled: false
+                      publishLocation: pipeline
+
+          # Job - Component detection
+          - ${{ if eq(parameters.publish, true) }}:
+              - job: CG
+                displayName: Component Detection
+                pool: Small-1ES
```
Comment (alexvy86) [inline tools/pipelines/templates/build-npm-package.yml:588]: Let's restore the use of `{{ parameters.pollCG }}` here. With this change the parameter would become unnecessary (I'd argue it already is unnecessary because we don't override its default anywhere) but seems like we didn't clean that up in the "master" migration PR, so better handle this separately and for now we can keep using it here. Just keep the indentation fix.

File: tools/pipelines/build-api-markdown-documenter.yml [hunk 4]
```diff
@@ -34,12 +34,12 @@ trigger:
     - .prettierignore
     - tools/api-markdown-documenter
     - tools/pipelines/build-api-markdown-documenter.yml
-    - tools/pipelines/templates/build-npm-package.yml
+    - tools/pipelines/templates/1ES/build-npm-package.yml
     - tools/pipelines/templates/include-set-package-version.yml
-    - tools/pipelines/templates/include-vars.yml
+    - tools/pipelines/templates/1ES/include-vars.yml
```
Comment (alexvy86) [inline tools/pipelines/build-api-markdown-documenter.yml:39]: Tied to a comment above, if we undo the change to include-vars we should revert this to use the non-1ES path, same for the other trigger below (line 62), and for the actual use of the template (line 69).

File: tools/pipelines/templates/include-policy-check.yml [hunk 5]
```diff
@@ -28,14 +28,14 @@ stages:
...
+  pool: Small-1ES
   jobs:
   - job:
     displayName: Run checks
     steps:
-    - template: include-use-node-version.yml
+    - template: : /tools/pipelines/templates/include-use-node-version.yml@self
 
-    - template: include-install-pnpm.yml
+    - template: : /tools/pipelines/templates/include-install-pnpm.yml@self
```
[the remark this answers is not in the document]
Comment (seanimam) [reply tools/pipelines/templates/include-policy-check.yml]: Fixed.

File: tools/pipelines/templates/include-vars.yml [hunk 6]
```diff
@@ -65,8 +65,6 @@ variables:
         eq(variables.shouldPublish, true)
       )
     )}}
-- name: componentDetection
-  value: ${{ variables.publish }}
```
[the remark this answers is not in the document]
Comment (seanimam) [reply tools/pipelines/templates/include-vars.yml]: Reverted the file and removed references to the 1ES version.

File: tools/pipelines/templates/build-npm-package.yml (hunk 3 above)
Comment (seanimam) [reply tools/pipelines/templates/build-npm-package.yml:588]: Updated, LMK if that looks right.

File: tools/pipelines/build-api-markdown-documenter.yml (hunk 4 above)
Comment (seanimam) [reply tools/pipelines/build-api-markdown-documenter.yml]: Updated.

File: tools/pipelines/templates/build-npm-package.yml (hunk 3 above)
Comment (seanimam) [reply tools/pipelines/templates/build-npm-package.yml]: did you mean to say this "poolCG" or pollCG?

Comment (alexvy86) [reply tools/pipelines/templates/build-npm-package.yml:588]: No, if you look at the diff the line to update was 588. We moved from a parameter for the pool name, to hardcoding Small-1ES. Line 585 needs to stick with `parameters.publish`. And my bad, typo in the previous comment, I think it's `poolCG`, not `pollCG` (you can confirm by looking at the parameters for the file).

Comment (seanimam) [reply tools/pipelines/templates/build-npm-package.yml]: Updated.
